vector-im/element-android
# ANR when asking to select the notification method

## Type of change


## Content

Refactoring most of the logic for registration of push notification distributor. Adding unit tests on the new code.

The previous code was a mix of UI and domain layer which lead to non necessary complexity and some bugs like ANR, crash or unexpected behavior.

## Motivation and context

Closes #7653 

## Screenshots / GIFs





## Tests



Both FDroid and Gplay version should be tested.
- Install Ntfy app
- Login to Element
- Check the distributor choice is asked when landing the first time in the home screen
- Check the notifications are well received in the different choices (background sync, Google services, Ntfy)
- Go to Settings -> Notifications
- Disable Notifications for the device
- Check the notifications are no more received
- Enable the notifications in settings
- Check distributor is asked
- Change the notification method in Settings -> Notifications
- Check notifications are still received after change of the method

## Tested devices

- OS version(s): Android 11

## Checklist
